src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -121,4 +220,82 @@ export class OsdListComponent implements OnInit {
   configureClusterAction() {
     this.bsModalRef = this.modalService.show(OsdFlagsModalComponent, {});
   }
+
+  showConfirmationModal(markAction: string, onSubmit: (id: number) => Observable<any>) {
```

Comment:
Could you please rename it?

```suggestion
  showActionModal(action: string, onSubmit: (id: number) => Observable<any>) {
```

Reply:
A modal can have an action or just be informational. If it provides an action, it may or may not be a confirmation. If it's a crucial operation, it'll be a critical confirmation. 
These dialogs are either confirmation dialogs or such which require additional confirmation for critical operations.

This is the reason why I think `confirmation` and `criticalConfirmation` are a perfect match here, as their description hits the nail on the head. They open (show) confirmation and critical confirmation dialogs (modals). 

I personally think `action` is way to general in this case and can't easily find an example where it might fit, except maybe as the parent class for all those modals.